components/motor/gpio/basic.go

```diff
@@ -175,10 +175,20 @@ func (m *Motor) turnOff(ctx context.Context, extra map[string]interface{}) error
 // Anything calling setPWM MUST lock the motor's mutex prior.
 func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]interface{}) error {
 	var errs error
+
 	powerPct = math.Min(powerPct, m.maxPowerPct)
 	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+	if math.Abs(powerPct) > m.minPowerPct {
+		powerPct = math.Max(powerPct, m.minPowerPct)
```

Comment:
probs can remove this line then

Reply:
oh wait I didn't see it was the absolute value. I think this issue will still be there. if the power is -0.5, the if statement is true, but then the max between -0.5 and 0 is still 0

Reply:
working on it. most test passing now :)